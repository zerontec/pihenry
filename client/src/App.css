/* .App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
 */

 :root{
  --gold1: #ffc107;
  --gold2: #f44336;
  --gray: #e3f2fd;
  --gray2: #f7d9d990;
  --black1: #1A1B1F;
  --black2: #30353B;
  --shadow: 0 8px 8px;
  --color_shadow: #00000033;
}

*{
  margin: 0;
  padding:0;
  box-sizing: border-box;
}

body{
  font-family: 'Roboto', sans-serif;
}
.container{
  width: 90%;
  max-width: 1200px;
  margin: 0 auto;
  overflow: hidden;
  padding:100px 0;
  height: auto;
}

.container--modifier{
  width: 60%;
}