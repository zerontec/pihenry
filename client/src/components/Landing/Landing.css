.hero{
  width: 100%;
  min-height: 100vh;
  background: linear-gradient(180deg, var(--black2) 0%, var(--black1) 100%);
}

.nav__hero{
  color: #f9f7f7;
  height: 70px;
  border-bottom: 1px solid rgba(255,255,255, 0.1);
  background-color: #e91e63;
}

.nav__container{
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: inherit;
  padding: 0;
  background-color: #e91e63;
 

 
}

.logo__name{
  font-size: 24px;
  color: white;
}

.link{
  display: inline-block;
  color: inherit;
  text-decoration: none;
  font-weight: 300;
  padding: 10px;
  color: white;
}

.link--active{
  color: var(--gold1);
  font-weight: 700;
}

.point{
  color: var(--gold1);
}

.hero__main{
  height: calc(100vh - 70px);
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding:0;
}

.mockup{
  object-fit: cover;
  width: 48%;
  max-height: 90vh;
}

.hero__textos{
  width: 48%;
}

.title{
  font-size: 56px;
  font-weight: 300;
  color: #fff;
}

.title--active{
  color: var(--gold1);
  font-weight: 700;
}

.copy{
  font-size: 24px;
  color: var(--gray);
  text-shadow: var(--shadow)  var(--color_shadow);
  margin: 15px 0;
}

.copy__active{
  font-weight: 700;
}


.cta{
  padding: 20px 30px;
  display: inline-block;
  background: linear-gradient(180deg, #F7CE90 0%, #F4BD76 100%);
  text-decoration: none;
  color: var(--black2);
  font-weight: 700;
  box-shadow: var(--shadow) var(--color_shadow);
  border-radius: 6px;
}



#content {
  background: #fff  ; /* Para Internet Explorer */
  background: rgba(255,255,255,.5)  ;
  border-radius: 10px;
  box-shadow: 5px 5px 10px rgb(200,200,200)  ;
  font-size: 2rem;
  margin: 0 auto;
  padding: 50px 0;
  text-align: center;
  text-shadow: 2px 2px 4px rgb(255, 255, 255)  ;
  width: 80%;
}

#foo {
  display: none;
  position: absolute;
  top: 20%;               
  width: 100%;
}

#video-background {
  height: 100%;
  position: fixed;
  width: 100%;
  z-index: -100;
}

