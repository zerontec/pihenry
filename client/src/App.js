import './App.css';
import {BrowserRouter as Router, Route, Switch }from 'react-router-dom';
import Landing from './components/Landing/Landing';
import Home from './components/Home/Home';
import Recipe from './components/Recipe/Recipe';
import Detail from './components/Detail/Detail';
import Contact from './components/contact/Contact';





function App() {
  return (
    <Router>
     
    <div className="App">
<Switch>
<Route exact path='/' component={Landing}/>
     <Route exact path='/home' component={Home}/>
     <Route exact path='/home:id' component={Detail}/>
     <Route exact path='/recipe' component={Recipe}/>
     <Route exact path='/contact' component={Contact}/>

      </Switch>
    </div>
    </Router>
  );
}

export default App;
