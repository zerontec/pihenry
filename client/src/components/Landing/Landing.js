import React from 'react';
import './Landing.css'
import { Link } from 'react-router-dom';
import video from './video.mp4'



export default function Landing(){



return(

    <div>

<header className="hero">
    

    <section className="container hero__main">  
        <div className="hero__textos">
            <h1 className="title">La mejor <span className="title--active">Experiencia  Vivela.</span></h1> 
            <p className="copy">Conoce las Mejores Recetas <span className="copy__active">Dietas de Alto nivel </span></p>
            <Link to='/Home' className="cta">Vamos</Link>
          
        </div>

    
        <video id='video-background' autoPlay loop muted>
    <source src={video} type='video/mp4' />
</video>
{/*         <video id="video-background" width="560" height="315" autoplay preload muted>
  <source src="./videos/video.mp4" type='video/mp4; codecs="avc1,mp4a"' />
  <source src="video.ogv" type='video/ogg; codecs="theora,vorbis"' />
  <source src="video.webm" type='video/webm; codecs="vp8,vorbis"' />
</video> */}
       {/*  <img src="./videos/comida.jpgs" className="mockup" alt='imagen comida'/> */}
    </section>
</header>


    </div>
)


}