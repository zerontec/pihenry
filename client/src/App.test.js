import { render, screen } from '@testing-library/react';
import App from './App';
import Landing from './components/Landing/Landing'
import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'



describe('Pruebas en <Landing />', () => {
  test('<Landing /> se renderiza bien', () => {
      const titulo = 'Esto es una Pruaba';
      const jsxLanding = shallow(<Landing titulo={titulo}/>);
      expect(jsxLanding).toMatchSnapshot();
  })

});
/* 
test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();

});
 */