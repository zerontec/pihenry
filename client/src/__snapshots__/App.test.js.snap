// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Landing /> <Landing /> se renderiza bien 1`] = `
<div>
  <header
    className="hero"
  >
    <section
      className="container hero__main"
    >
      <div
        className="hero__textos"
      >
        <h1
          className="title"
        >
          La mejor 
          <span
            className="title--active"
          >
            Experiencia  Vivela.
          </span>
        </h1>
        <p
          className="copy"
        >
          Conoce las Mejores Recetas 
          <span
            className="copy__active"
          >
            Dietas de Alto nivel 
          </span>
        </p>
        Esto es una Pruaba
      </div>
      <video
        autoPlay={true}
        id="video-background"
        loop={true}
        muted={true}
      >
        <source
          src="video.mp4"
          type="video/mp4"
        />
      </video>
    </section>
  </header>
</div>
`;
